.github/workflows/static-analysis-js.yml

```diff
@@ -0,0 +1,104 @@
+        description: Authentication for the private npm registry.
+        required: false
+      GITHUB_USER_EMAIL:
+        description: Email address for the GitHub user configuration.
+        required: false
+      GITHUB_USER_NAME:
+        description: Username for the GitHub user configuration.
+        required: false
+      GITHUB_USER_SSH_KEY:
+        description: Private SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: false
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  static-analysis-js:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    env:
+      NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+      NODE_CACHE_MODE: ''
+      GITHUB_USER_EMAIL: ${{ secrets.GITHUB_USER_EMAIL }}
+      GITHUB_USER_NAME: ${{ secrets.GITHUB_USER_NAME }}
+      GITHUB_USER_SSH_KEY: ${{ secrets.GITHUB_USER_SSH_KEY }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up custom environment variables
+        env:
+          ENV_VARS: ${{ secrets.ENV_VARS }}
+        if: ${{ env.ENV_VARS }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+
+      - name: Set up SSH
+        if: ${{ env.GITHUB_USER_SSH_KEY != '' }}
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ env.GITHUB_USER_SSH_KEY }}
+
+      - name: Set up Git
+        run: |
+          git config --global user.email "${{ env.GITHUB_USER_EMAIL }}"
+          git config --global user.name "${{ env.GITHUB_USER_NAME }}"
+
+      - name: Set up node cache mode
+        run: |
+          if [ -f "${GITHUB_WORKSPACE}/package-lock.json" ] || [ -f "${GITHUB_WORKSPACE}/npm-shrinkwrap.json" ]; then
+            echo "NODE_CACHE_MODE=npm" >> $GITHUB_ENV
+          else
+            echo "No lock files found"
+          fi
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: ${{ env.NODE_CACHE_MODE }}
+
+      - name: Install dependencies
+        env:
+          ARGS: ${{ env.NODE_CACHE_MODE == 'npm' && 'ci' || 'install' }}
+        run: ${{ format('npm {0}', env.ARGS) }}
+
+      - name: Run tsc
+        run: |
+          ./node_modules/.bin/tsc --noEmit --skipLibCheck --pretty false |
+            while read -r line
+              do
+                if [[ $line =~ ([0-9a-z\/\.]+)[\(]([0-9]+)[\,]([0-9]+)[\)].*(TS.*) ]]; then
+                echo "::error file=${BASH_REMATCH[1]},line=${BASH_REMATCH[2]},col=${BASH_REMATCH[3]}::${BASH_REMATCH[4]}"
+              fi
+            done
+          exit "${PIPESTATUS[0]}"
```

Comment:
I did some checks, and I am wondering whether this is necessary.
1. Shouldn't this work out of the box because `actions/setup-node`  adds a [problem matcher](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md) for TypeScript already? See https://github.com/actions/setup-node/blob/97ca147735c170fb35096b39ef17a0fc5d9270ac/src/main.ts#L69, where it adds https://github.com/actions/setup-node/blob/97ca147735c170fb35096b39ef17a0fc5d9270ac/.github/tsc.json.
2. If this does not work for whatever reason, can we do something like this?
   ```yml
   - name: Set up problem matchers for tsc
     run: echo "::add-matcher::${{ runner.tool_cache }}/tsc.json"
   ``` 
   Similar to here ([documentation](https://github.com/shivammathur/setup-php#phpunit)): https://github.com/inpsyde/reusable-workflows/blob/4a4c5f3046984a87a046d5ddb3c1e98da60dc123/.github/workflows/tests-unit-php.yml#L55-L56
3. Alternatively, can we do it like here?
   ```yml
   - name: Set up problem matchers for tsc
     run: |
        curl -s -o tsc-matcher.json https://raw.githubusercontent.com/actions/setup-node/main/.github/tsc.json
        echo "::add-matcher::$(GITHUB_WORKSPACE)/tsc-matcher.json"
   ```
   Similar to here: https://github.com/inpsyde/reusable-workflows/pull/158/files#diff-20ce74f39298d8724bf08ab59184d84e7a21326dcaa25bd2b5c5e3493b0727e0R17-R20.

Reply:
Yeah, it works out of the box. Thanks a lot for your insight! :rocket: